pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,403 @@
+#!/bin/bash
+# (return 0 2>/dev/null) && sourced=1 || sourced=0
+
+INNER_SCRIPTS_PATH="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+# shellcheck source=bin/common.bash
+source "${INNER_SCRIPTS_PATH}/../funcs.sh"
+
+pushd "${CK8S_CONFIG_PATH}" || exit
+adminPassword=$(sops -d secrets.yaml | yq4 '.opensearch.adminPassword')
+fluentdPassword=$(sops -d secrets.yaml | yq4 '.opensearch.fluentdPassword')
+opsDomain=$(yq4 '.global.opsDomain' common-config.yaml)
+popd || exit
+
+
+function opensearch_check_help(){
+    printf "%s\n" "[Usage]: test sc opensearch [ARGUMENT]";
+    printf "\t%-25s %s\n" "--cluster-health" "Get cluster health";
+    printf "\t%-25s %s\n" "--snapshot-status" "Check snapshot status";
+    printf "\t%-25s %s\n" "--breakers" "Check if circuit breakers have been triggered";
+    printf "\t%-25s %s\n" "--indices" "Check if there are any missing indices";
+    printf "\t%-25s %s\n" "--aliases" "Check if each aliase has a write index";
+    printf "\t%-25s %s\n" "--mappings" "Check mappings/fields count & limit";
+    printf "\t%-25s %s\n" "--user-roles" "Check configured user roles";
+    printf "\t%-25s %s\n" "--ism" "Check ISM";
+    printf "\t%-25s %s\n" "--object-store-access" "Check object store access";
+    printf "\t%-25s %s\n" "--fluentd" "Check that fluentd can connect to opensearch";
+    printf "%s\n" "[NOTE] If no argument is specified, it will go over all of them.";
+
+    exit 0;
+}
+
+function sc_opensearch_checks(){
+    if [[ ${#} == 0 ]]; then
+        check_opensearch_cluster_health
+        check_opensearch_snapshots_status
+        check_opensearch_breakers
+        check_opensearch_indices
+        check_opensearch_aliases
+        check_opensearch_mappings
+        check_opensearch_user_roles
+        check_opensearch_ism
+        check_object_store_access
+        check_fluentd_connection
+        exit 0
+    fi
+    while [[ ${#} -gt  0 ]]; do
+      case ${1} in
+          --cluster-health )
+              check_opensearch_cluster_health
+          ;;
+          --snapshot-status )
+              check_opensearch_snapshots_status
+          ;;
+          --breakers )
+              check_opensearch_breakers
+          ;;
+          --indices )
+              check_opensearch_indices
+          ;;
+          --aliases )
+              check_opensearch_aliases
+          ;;
+          --mappings )
+              check_opensearch_mappings
+          ;;
+          --user-roles )
+              check_opensearch_user_roles
+          ;;
+          --ism )
+              check_opensearch_ism
```

Comment:
Could you add a check to see if ISM is rolling over write indices as it should?
With `GET _plugins/_ism/policies/authlog` or from the config you should be able to get how old the index may be before it is rolled over. Then you can just inspect the creation date on the write indices to check if they are older than that.

Reply:
Sure, it's added now